### DISCUSSION TITLE
Add custom field to pivot table

### DISCUSSION THREAD

**Discussion by dbrab** [spatie/laravel-permission#1897] [Ideas]
I would like to know, if you have considered adding an additional field to the pivot table, for example model-permission.

I have users to which certain days I need to disable access, control the amount of daily and monthly posts, comments, updates, etc. that they can do, I would like to be able to control this metadata from the same permission when assigning it to a user.

Example:

```

// id = 1 -> name Access-To Panel guard_name = yyy options = json_encode ({days: ['Monday'],})
$ permission1 =

// id = 2 name = "Create Post" options = json_encode ({countMaxTotal: 100, countMaxDay: 10, countMaxWeek: 40})
$ permission2 =
$ user-> givePermissionTo ([$ permission1, $ permission2]);


$ permissions = User .... permissions () -> get ()

$ permissions [0] -> pivot-> options
// json_decode (options).
```

Something like this would help me to better control this metadata, I currently have additional tables for it, but it seems not very flexible, since I can reuse the pivot tables that spatie already has for the same.

This can be extended for roles and teams.

**Comment by erikn69** [comment]
You can overwrite migration and trait methods, and do everything you need.
Overwrite example: https://github.com/spatie/laravel-permission/discussions/1895#discussioncomment-1523163
Migration example: https://github.com/spatie/laravel-permission/discussions/1827#discussioncomment-1234029
Only overwrite `hasPermissionTo`
```php
    public function hasPermissionTo($permission, $guardName = null): bool
    {
        $permissionClass = $this->getPermissionClass();

        if (is_string($permission)) {
            $permission = $permissionClass->findByName($permission, $this->getDefaultGuardName());
        }

        if (is_int($permission)) {
            $permission = $permissionClass->findById($permission, $this->getDefaultGuardName());
        }

        if ($permission instanceof Permission && $assigned = $this->permissions->where('id', $permission->id)) {
            $options = $assigned->pivot->options;
            if (/* $options don't pass validations */) {
                   return false;
            }
        }

        return $this->hasPermissionToFromParent($permission, $guardName);
    }
```
